### PR TITLE
posts query filters

### PR DESCRIPTION
**Pull request by Schparky**

### CODE REVIEW

File: application/models/post.go [hunk 1]
```diff
@@ -811,28 +810,37 @@ func (p *Posts) FindByUser(ctx context.Context, user User) error {
...
+	AND status not in (?, ?)`
+
+	args := []interface{}{user.ID, PostVisibilityAll, PostVisibilityTrusted, PostStatusRemoved,
+		PostStatusCompleted}
+
+	if searchText != nil {
+		selectClause = selectClause + " AND (LOWER(title) LIKE ? or LOWER(description) LIKE ?)"
+		likeText := "%" + strings.ToLower(*searchText) + "%"
+		args = append(args, likeText)
+		args = append(args, likeText)
```
Comment (Baggerone) [inline application/models/post.go]: I think you can just include this second `likeText` in the first append.

File: application/models/post.go [hunk 2]
```diff
@@ -811,28 +810,37 @@ func (p *Posts) FindByUser(ctx context.Context, user User) error {
...
+	}
 
-	return DB.
-		Scope(scopeUserOrgs(user)).
-		Scope(scopeNotCompleted()).
-		Where(where, pType, contains, contains).
-		Order("created_at desc").
-		All(p)
+	for i := range posts {
+		*p = append(*p, posts[i])
```
Comment (Baggerone) [inline application/models/post.go]: It looks like this assumes that p begins as an empty slice. Should we error if it's not?

File: application/models/post.go [hunk 3]
```diff
@@ -1007,3 +1015,28 @@ func (p *Post) Meeting() (*Meeting, error) {
 
 	return &meeting, nil
 }
+
+// FilterDestination returns a list of all posts with a Destination near the given location. The database is not
+// touched.
+func (p Posts) FilterDestination(location Location) Posts {
+	filtered := make(Posts, 0)
+	for i := range p {
+		_ = DB.Load(&p[i], "Destination")
```
Comment (Baggerone) [inline application/models/post.go]: Can this go before the `for loop` and act on all the posts at once? (Same for next function)

File: application/gqlgen/meeting.go [hunk 4]
```diff
@@ -236,7 +236,7 @@ func (r *mutationResolver) CreateMeeting(ctx context.Context, input meetingInput
 		return nil, reportError(ctx, err, "CreateMeeting.Unauthorized", extras)
 	}
 
-	location := convertGqlLocationInputToDBLocation(*input.Location)
+	location := convertLocation(*input.Location)
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/gqlgen/meeting.go]: It's an unfortunate side-effect of having the same struct for both update and create. Unless I'm misunderstanding your comment.

File: application/gqlgen/schema.graphql [hunk 5]
```diff
@@ -12,37 +12,25 @@ type Query {
     Posts, aka Requests. With no parameters supplied, all posts visible to the authenticated user are returned. Filter
     parameters only remove from this default list and never include posts that are not visible to the authenticated
     user. For posts associated with a `User` or `Meeting`, use the `posts` field on `User` and `Meeting`.
-    NOT YET IMPLEMENTED: `destination`, `origin`, `searchText`
     """
     posts(
         """
-        NOT YET IMPLEMENTED --
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/gqlgen/schema.graphql]: Yep

File: application/models/post.go (hunk 1 above)
Comment (Schparky) [reply application/models/post.go]: Good.

File: application/models/post.go (hunk 2 above)
Comment (Schparky) [reply application/models/post.go]: Probably, yeah.

Comment (Schparky) [reply application/models/post.go]: Or what if I just empty it out instead of error it?

Comment (Baggerone) [reply application/models/post.go]: That sounds good.

File: application/models/post.go (hunk 3 above)
Comment (Schparky) [reply application/models/post.go]: Good thought. It doesn't combine into one DB query, but a little simpler nonetheless.

File: application/models/user.go [hunk 6]
```diff
@@ -746,3 +746,13 @@ func (u User) GetLanguagePreference() string {
 func (u *User) GetRealName() string {
 	return strings.TrimSpace(u.FirstName + " " + u.LastName)
 }
+
+// HasOrganization returns true if the user has one or more organization connections
+func (u *User) HasOrganization() bool {
+	var c Count
+	err := DB.RawQuery("SELECT COUNT(*) FROM user_organizations WHERE user_id = ?", u.ID).First(&c)
+	if err != nil || c.N == 0 {
```
Comment (Baggerone) [inline application/models/user.go]: Maybe log real errors, so we can troubleshoot if this doesn't work for some reason?

Comment (Schparky) [reply application/models/user.go]: OK

**Pull request merged**